### PR TITLE
Profiling test

### PR DESCRIPTION
**Pull request by julibert**
This PR adds a profiling test for Agent.

### CODE REVIEW

File: test/profiling/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Profiling test
+
+Profiling test is composed of tree components:
```
Comment (jamoralp) [inline test/profiling/README.md:3]:
```suggestion
Profiling test is composed of three components:
```

File: test/profiling/README.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+* A minimal Agent with just `UAGENT_CED_PROFILE` enable.
+* A publisher developed with the Client library.
+* A subscriber developed with the Client library.
+
+## Why?
+
+There are some scenarios where the Agent must run over low resource devices, so it is needed to characterize the minimum requirements of the Agent in terms of memory footprint.
+In that sense, the Agent profiling with the lowest memory footprint is the one that only use the `CedMiddleware` implementation without logger, cli, and p2p.
+Such configuration could be address disabling all the Agent`s profiles except the `UAGENT_CED_PROFILE`.
```
Comment (jamoralp) [inline test/profiling/README.md:13]:
```suggestion
Such configuration could be address disabling all the Agent's profiles except the `UAGENT_CED_PROFILE`.
```

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS%20Manual%20Linux/108/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS%20Manual%20Windows/127/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS%20Manual%20Linux/109/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS%20Manual%20Windows/128/)

**Pull request merged**